distributed/scheduler.py

```diff
@@ -7123,22 +7163,27 @@ async def retire_workers(
         --------
         Scheduler.workers_to_close
         """
+        if names is not None and workers is not None:
+            raise TypeError("names and workers are mutually exclusive")
+        if (names is not None or workers is not None) and kwargs:
+            raise TypeError(
+                "Parameters for workers_to_close() are mutually exclusive with "
+                f"names and workers: {kwargs}"
+            )
```

Comment:
OT: Is there a specific reason you prefer `TypeError` over `ValueError`? I guess an argument could made for either.

Reply:
Unexpected parameters to a function raise TypeError natively.
In my head, the fact that the function *can* accept both names and workers at the same time is a technicality - what counts are the overloaded signatures, which can't. In other words: it would make  sense to have 3 separate functions,
``retire_workers_by_address(addresses)``, ``retire_workers_by_name(names)`` and ``retire_workers_auto()``. We have a single function with 3 different signatures because it's slightly nicer UX. But otherwise it should behave like 3 separate functions.

Reply:
Thanks for explaining your thoughts, this makes sense to me. I approached it from the other end, where the arguments all have allowed types, but the combination of values is invalid.